### PR TITLE
libc: unix-socket: separate test cases

### PR DESCRIPTION
**Pull request by adamdebek**
JIRA: CI-404

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: libc/test.yaml [hunk 1]
```diff
@@ -67,7 +67,7 @@ test:
       execute: test-libc-unix-socket
       targets:
         # Excluded due to long execution time
-        exclude: [armv7m4-stm32l4x6-nucleo]
+        exclude: [armv7m4-stm32l4x6-nucleo, armv7a9-zynq7000-qemu]
```
Comment (mateusz-bloch) [inline libc/test.yaml:70]: I see, the issue with the tests taking too long relates to the test cases:` unix_msg_data_and_fd` and `unix_msg_fork`. Could we split each of these into two separate test cases, for example one for `SOCK_STREAM` and one for `SOCK_DGRAM`? This way, we could run them on every target, given that the issue here is simply that the Nucleo and Zynq-QEMU targets are slow.

File: libc/socket/unix-socket.c [hunk 2]
```diff
@@ -405,9 +405,15 @@ void unix_msg_data_only(int type)
 	close(fd[1]);
 }
 
-TEST(test_unix_socket, unix_msg_data_only)
+
+TEST(test_unix_socket, unix_stream_sock_send_data)
```
Comment (mateusz-bloch) [inline libc/socket/unix-socket.c:409]: Not that test case? From what I see, these should be changed:  `unix_msg_data_and_fd` & `unix_msg_fork` :

Zynq-qemu
image

stm32:
image

**Review by mateusz-bloch — COMMENTED** [review]
Also change the commit message

File: libc/test.yaml [hunk 3]
```diff
@@ -67,7 +67,6 @@ test:
       execute: test-libc-unix-socket
       targets:
         # Excluded due to long execution time
-        exclude: [armv7m4-stm32l4x6-nucleo]
```
Comment (mateusz-bloch) [inline libc/test.yaml:70]: delete `# Excluded due to long execution time`

File: libc/socket/unix-socket.c [hunk 4]
```diff
@@ -475,9 +476,14 @@ void unix_msg_data_and_fd(int type)
 }
 
 
-TEST(test_unix_socket, unix_msg_data_and_fd)
+TEST(test_unix_socket, unix_stream_sock_send_and_fd)
```
Comment (damianloew) [inline libc/socket/unix-socket.c:479]: I think that we can remove this `unix` prefix as it's mentioned in the name of test group. Then we can name it `msg_data_and_fd_strem`. The current name is misleading - send and file descriptor, send what (?), also it's not only sending, but receiving too. IMHO it's better to keep the previous convention, where we have test cases like `data_only` and `data_and_fd`

Comment (damianloew) [reply libc/socket/unix-socket.c:479]: We can name the same way the rest of changed cases.

**Review by damianloew — CHANGES_REQUESTED** [review]
Thanks for the PR, have you made sure that fails are no longer seen?

**Comment by mateusz-bloch** [comment]
> Thanks for the PR, have you made sure that fails are no longer seen?

Yes,  from what I remember, in the worst-case scenario there's about a +-12-second safety margin.

**Review by damianloew — APPROVED** [review]
LGTM, Could you please update submodules after a merge?

**Pull request merged**